include/c_utilities/filesystem.h

```diff
@@ -0,0 +1,107 @@
+// Copyright 2017 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef C_UTILITIES__FILESYSTEM_H_
+#define C_UTILITIES__FILESYSTEM_H_
+
+#if __cplusplus
+extern "C"
+{
+#endif
+
+#include <stdbool.h>
+
+#include "c_utilities/macros.h"
+#include "c_utilities/visibility_control.h"
+
+/// Return current working directory.
+/**
+ * \param[in] buffer Allocated string to store current directory path to
+ * \param[in] max_length maximum length to be stored in buffer
+ * \return bool True if success False otherwise
+ */
+C_UTILITIES_PUBLIC
+C_UTILITIES_WARN_UNUSED
+bool
+utilities_get_cwd(char * buffer, size_t max_length);
```

Comment:
For all these functions which take in a `char *` of some type or another. What happens if I pass `NULL`? I think it's worth just putting a in each docblock that says `False` is returned in that case, since that's the easiest solution.

A more extreme, but precise, solution would be to return a return code rather than a bool, so that the user could know if it failed because of the check or because of some other logistical issue, like a null pointer being passed.